street_level_crime/create_csv.py

```diff
@@ -7,13 +9,52 @@ def convert_to_csv(headers: tuple, data: list) -> csv:
     :Tuple headers: Contians the headers of the csv file
     :List data: Contains a list of row data in format List[List[String]]
     """
+    print("When entering a filename please don't add .csv to the file")
+    filename = input("Please enter a filename: ")
+    special_commands(filename)
     try:
-        with open('crime data.csv', 'w', newline='') as f:  # creates document and opens for writer object
-            csvWriter = csv.writer(f)  # creates writer object
-            csvWriter.writerow(headers)  # writes header
-            for i in range(0, len(data) - 1, 1):  # writes each row in array
-                csvWriter.writerow(data[i])  # writes from
-
+        with open(filename + '.csv', 'w+', newline='') as f:  # creates document and opens for writer object
+            csvWriter = csv.writer(f)
+            csvWriter.writerow(headers)
+            for i in range(0, len(data) - 1, 1):
+                csvWriter.writerow(data[i])
+            print("File Created:", filename + ".csv")
+            f.close()
+            print("Do you want to save to a custom path?")
+            print("[1] Yes")
+            print("[2] No")
+            CustomPathInput = input()
+            special_commands(CustomPathInput)
+            if CustomPathInput == "1":
+                custom_path(filename)
+            else:
+                print("Saved to default location: " + os.path.dirname(
+                    os.path.realpath(filename + ".csv")) + '\\' + filename + ".csv")
     except:
         print("Error writing to CSV")
         csvWriter.writerow("Error writing to CSV")
+
+
+def custom_path(filename):
+    try:
+        src = os.path.dirname(os.path.realpath(filename + ".csv")) + '\\' + filename + ".csv"
+        home = os.path.expanduser("~")
+        dst = home + "\\" + input("Please enter a directory: " + home + "\\")
+        special_commands(dst)
+        if os.path.exists(dst):
+            shutil.move(src, dst)
+            print("Saved to: " + dst)
+        else:
+            print("Saved to default location: " + src)
+    except:
+        print("Unable to save file to custom path")
+
+
+def special_commands(variable):
+    pass
+    # Link to Jack Gittoes' Code
```

Comment:
The commands stuff is a tricky one, Need to have a look at what jack has done with the UI but we'll find a unified way of doing this

Reply:
Waiting for Jack's new functions